fearless/Modules/Staking/StakingBalance/StakingBalanceInteractor+Subscription.swift

```diff
@@ -0,0 +1,173 @@
+import RobinHood
+import IrohaCrypto
+
+extension StakingBalanceInteractor {
+    func subscribeToPriceChanges() {
+        let updateClosure = { [weak self] (changes: [DataProviderChange<PriceData>]) in
+            if changes.isEmpty {
+                self?.presenter.didReceive(priceResult: .success(nil))
+            } else {
+                for change in changes {
+                    switch change {
+                    case let .insert(item), let .update(item):
+                        self?.presenter.didReceive(priceResult: .success(item))
+                    case .delete:
+                        self?.presenter.didReceive(priceResult: .success(nil))
+                    }
+                }
+            }
+        }
+
+        let failureClosure = { [weak self] (error: Error) in
+            self?.presenter.didReceive(priceResult: .failure(error))
+            return
+        }
+
+        let options = DataProviderObserverOptions(
+            alwaysNotifyOnRefresh: false,
+            waitsInProgressSyncOnAdd: false
+        )
+        priceProvider.addObserver(
+            self,
+            deliverOn: .main,
+            executing: updateClosure,
+            failing: failureClosure,
+            options: options
+        )
+    }
+
+    func subscribeToElectionStatus() {
+        guard electionStatusProvider == nil else {
+            return
+        }
+
+        guard let electionStatusProvider = try? providerFactory
+            .getElectionStatusProvider(chain: chain, runtimeService: runtimeCodingService)
+        else {
+            return
+        }
+
+        self.electionStatusProvider = electionStatusProvider
+
+        let updateClosure = { [weak self] (changes: [DataProviderChange<DecodedElectionStatus>]) in
+            if let electionStatus = changes.reduceToLastChange() {
+                self?.presenter.didReceive(electionStatusResult: .success(electionStatus.item))
+            }
+        }
+
+        let failureClosure = { [weak self] (error: Error) in
+            self?.presenter.didReceive(electionStatusResult: .failure(error))
+            return
+        }
+
+        let options = DataProviderObserverOptions(
+            alwaysNotifyOnRefresh: false,
+            waitsInProgressSyncOnAdd: false
+        )
+        electionStatusProvider.addObserver(
+            self,
+            deliverOn: .main,
+            executing: updateClosure,
+            failing: failureClosure,
+            options: options
+        )
+    }
+
+    func subsribeToActiveEra() {
+        let codingFactoryOperation = runtimeCodingService.fetchCoderFactoryOperation()
+
+        let activeEraWrapper: CompoundOperationWrapper<ActiveEraInfo?> =
```

Comment:
We need to use subscription to the active era so we handle the case when unbondings/reedemables changes after a user opens the screen